srvclient.go

```diff
@@ -53,11 +69,24 @@ func lookupSRV(hostname string) (*dns.Msg, error) {
 		m2.SetQuestion(fqdn, dns.TypeSRV)
 		res, _, err = c.Exchange(m2, server+":53")
 		if err == nil {
-			return res, nil
+			break
 		}
 	}
 
-	return nil, errors.New("no available nameservers")
+	if res == nil {
+		return nil, errors.New("no available nameservers")
+	}
+	if len(res.Answer) == 0 {
+		return nil, fmt.Errorf("No SRV records for %q", hostname)
+	}
+
+	ans := make([]*dns.SRV, len(res.Answer))
+	for i := range res.Answer {
+		if ansSRV, ok := res.Answer[i].(*dns.SRV); ok {
+			ans[i] = ansSRV
+		}
+	}
```

Comment:
should you add a check at this point to make sure `len(ans) > 0`? If so you probably don't even need the one above this for loop